scripts/player.gd

```diff
@@ -86,20 +86,23 @@ func equip_bomb(idx):
 func set_nearby_npc(npc):
 	nearby_npc = npc
 
-func able_to_climb(stairs, dir, act):
+func able_to_climb(stairs):
 	return !climbing and can_climb and stairs.get_cellv(stairs.world_to_map(self.get_pos())) != -1 \
-		and (dir != -1 and dir != DIR.RIGHT and dir != DIR.LEFT) \
-		and not ((dir == DIR.DOWN or dir == DIR.DOWN_RIGHT or dir == DIR.DOWN_LEFT) \
-		and !ground.get_overlapping_bodies().empty()) and act != ACT.CAMERA
+		and (!input.is_direction_held(DIR.INVALID) and !input.is_direction_held(DIR.RIGHT) and !input.is_direction_held(DIR.LEFT)) \
+		and not ( \
+				( \
+					input.is_direction_held(DIR.DOWN) or \
+					input.is_direction_held(DIR.DOWN_RIGHT) or \
+					input.is_direction_held(DIR.DOWN_LEFT) ) \
+				and !ground.get_overlapping_bodies().empty() ) \
+		and not input.is_action_held(ACT.CAMERA)
```

Comment:
This formatting is unusual. When we break a line after an `(`, we usually also break a line before the `)`. Either break the line in both, or don't break at all.